plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -707,10 +621,12 @@ private void fillNextEntries()
 
             // process page
             for (int i = 0; i < extractors.size(); ++i) {
-                DeltaLakeTransactionLogEntry entry = extractors.get(i).getEntry(session, page.getBlock(i).getLoadedBlock(), pagePosition);
-                if (entry != null) {
-                    nextEntries.add(entry);
```

Comment:
We may want to restore this condition after #19588 + early exit in `buildAddEntry` method.